foxpuppet/windows/browser/tab_bar.py

```diff
@@ -0,0 +1,98 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/
+"""Creates tab_bar object to interact with the Firefox tabs and tabbar."""
+
+from selenium.webdriver.common.by import By
+
+from foxpuppet.region import Region
+
+
+class Tab_bar(Region):
+    """Representation of the tab bar which contains the tabs.
+
+    Args:
+        window (:py:class:`BaseWindow`): Window object this region appears in.
+        root
+            (:py:class:`~selenium.webdriver.remote.webelement.WebElement`):
+            WebDriver element object that serves as the root for the
+            region.
+
+    """
+
+    _new_tab_button_locator = (By.ID, 'new-tab-button')
+    _tabs_locator = (By.TAG_NAME, 'tab')
+
+    @property
+    def tabs(self):
+        """Return a list of tabs.
+
+        Returns: :py:class:`~foxpuppet.window.browser.tab_bar.Tab`
+
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            tabs = [self.Tab(self, el) for el in
+                    self.selenium.find_elements(*self._tabs_locator)]
+        # Assign handles
+        for tab, handle in zip(tabs, self.selenium.window_handles):
```

Comment:
I'll look into this. I was under the impression that it will be in order of tab opened since it is a list. 

Now thinking about the multiple window situation...technically, geckodriver would see it as the same session since you can have multiple windows per profile already. I don't know if there is a way to differentiate tabs from windows as far as ```window_handles``` goes. I think they are all in one. 

Its a little tricky because do we assume that a user testing this will only open one window? If there is no way to tell a window and a tab apart, unless we create a foxpuppet tracking object that has a ```dict``` object and can tell the user which is which depending on when they opened it.

Reply:
> do we assume that a user testing this will only open one window?

I wouldn't make this assumption. We're providing them with an API for opening new windows and now tabs.

Reply:
There is no order of window handles at all. Also note that window handles can change right now given if the tab is moved to a different content process (like remoteness change).

Chrome windows of Firefox are handled via `chrome_window_handles` but not sure if we pass those through geckodriver yet.

Reply:
Going along with whimboo, it seems as if it doesn't matter if you open a window or a tab, selenium sees that as a ```window_handle```.

We could either keep track of each event with a dict but would that be necessary? It seems as if we need an easy way to differentiate between a ```chrome``` window handle and a "Normal" window_handle.

Reply:
If you are going to open a new tab, you are interested in that tab aka `window_handle` and not the window aka `chrome_window_handle`. Also it doesn't matter for the test in which window this tab is located in.